### DISCUSSION TITLE
easyPrint filename

### DISCUSSION THREAD

**Discussion by jwevandijk** [Leaflet/Leaflet#8635] [Q&A]
I have a map with the easyPrint tool that works as expected. However I want the filename to be dependent on the center coordinate. What I have in my latest attempt is:

    var printer = L.easyPrint({
          tileLayer: null,
          tileWait: 4000,
          sizeModes: ['Current', 'A4Landscape', 'A4Portrait'],
          filename: 'kaart-europa.png',
          exportOnly: true,
          hideControlContainer: false,
          position: 'topright'
    }).addTo(map)

    map.on('dragend', function onDragEnd(){
        z = map.getZoom()
        center = map.getCenter()
        center_lon = center.lng.toFixed(5)
        center_lat = center.lat.toFixed(5)
        fn = 'kaart-europa-' + center_lon + 'E-' + center_lat + 'N-' + z
        console.log('fname: ' + fn)
        printer.filename = filename()
        console.log('printer: ' + printer.filename))
    });
When I drag the map I see that printer.filename has changed but when I click the print button in the map the original filename is used.
Obviously my understanding of the leaflet and js object model is limited.
Please help,
Janwillem

**Comment by IvanSanchez** [comment]
You're making newbie mistakes - you're defining `fn` as a string, but then you're using a `filename()` function call. I suggest you review https://developer.mozilla.org/en-US/docs/Learn/JavaScript

You're also assuming that items in the options dictionary passed to the `L.easyPrint` factory magically get converted into instance properties. They don't. In Leaflet stuff, initialization options reside in the `options` property. See e.g. https://github.com/rowanwins/leaflet-easyPrint/blob/e77e333078eaee07429ed740069abe20e57f10f5/src/index.js#L82
